git/fetch-git.yaml

```diff
@@ -0,0 +1,41 @@
+apiVersion: tekton.dev/v1alpha1
+kind: Task
+metadata:
+  name: fetch-git
+spec:
+  inputs:
+    params:
+    - name: url
+      description: git url to clone
+      type: string
+    - name: revision
+      description: git revision to checkout (branch, tag, sha, ref…)
+      type: string
+      default: master
+    - name: workingDirectory
+      description: working directory to clone
+      type: string
+      default: .
+    - name: submodules
+      description: init and fetch recursively submodules
+      type: string
+      default: "true"
+  steps:
+  - name: clone
+    image: gcr.io/tekton-releases/github.com/tektoncd/pipeline/cmd/git-init:latest
+    workingdir: /workspace/src/
+    command:
+    - /ko-app/git-init
+    args:
```

Comment:
Can we add the parameter `-depth` and default to `1` this way it can be use in `git clone --branch master --depth 1` and download 10KB instead of the whole history 100MB of the repo?
cc @vdemeester

Reply:
@csantanapr yeah we will add support for `--depth`. We need to update this yaml to use something else than `git-init`, using the `git` binary itself and `script` most likely.